File: 1207-unique-number-of-occurrences/solution.py
class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        s = {}
        for i in arr:
            if i not in s:
                s[i] = 1
            else:
                s[i] += 1

        a = {}
        for i in s.values():
            a[i] = True
        
        return len(s.keys()) == len(a.keys())

File: 1207-unique-number-of-occurrences/solution.go
package main

func uniqueOccurrences(arr []int) bool {
	s := make(map[int]int)
	for _, i := range arr {
		s[i]++
	}

	a := make(map[int]bool)
	for _, k := range s {
		a[k] = true
	}

	return len(a) == len(s)
}
